### PR TITLE
docs: PBTS synchrony issues runbook

### PR DESCRIPTION
**Pull request by williambanfield**
closes: #7756 

# What does this pull request change?
This pull request adds a new runbook for operators enountering errors related to the new Proposer-Based Timestamps algorithm. The goal of this runbook is to give operators a set of clear steps that they can follow if they are having issues producing blocks because of clock synchronization problems. 
This pull request also renames the `*PrevoteDelay` metrics to drop the term `MessageDelay`. These metrics provide a combined view of `message_delay` + `synchrony` so the name may be confusing.
# Questions to reviewers
* Are there ways to make the set of steps clearer or are there any pieces that seem confusing?

### CODE REVIEW

File: docs/tools/debugging/proposer-based-timestamps-runbook.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+### Updating SynchronyParams
+
+The `SynchronyParams` are `ConsensusParameters` which means they are set and updated
+by the application running alongside Tendermint. Updates to these parameters must
+be passed to the application during the `FinalizeBlock` ABCI method call.
+
+If the application was built using the CosmosSDK, then these parameters can be updated
+programatically using a governance proposal. For more information, see the [CosmosSDK
+documentation]().
```
Comment (williambanfield) [inline docs/tools/debugging/proposer-based-timestamps-runbook.md:203]: My searching has yet to turn up this documentation.

Comment (williambanfield) [reply docs/tools/debugging/proposer-based-timestamps-runbook.md:203]: To clarify, the documentation I'm hoping to link here will be just on what a governance proposal is, not necessarily the specific CosmosSDK documentation on PBTS.

File: docs/tools/debugging/proposer-based-timestamps-runbook.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+### Checking Synchrony
+
+To determine the currently configured `SynchronyParams` for your network, issue a
+request to your node's RPC endpoint. For a node running locally with the RPC server
+exposed on port `26657`, run the following command:
+
+```
+# curl localhost:26657/consensus_params
+```
```
Comment (creachadair) [inline docs/tools/debugging/proposer-based-timestamps-runbook.md:178]:
```suggestion
```shell
curl localhost:26657/consensus_params
```
```

File: docs/tools/debugging/proposer-based-timestamps-runbook.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+```json
+{
+  "precision": "500000000",
+  "message_delay": "3000000000"
+}
+```
+
+The `precision` and `message_delay` values returned are listed in nanoseconds.
+Remember, `tendermint_consensus_quorum_prevote_delay` is listed in seconds.
```
Comment (creachadair) [inline docs/tools/debugging/proposer-based-timestamps-runbook.md:190]:
```suggestion
The `precision` and `message_delay` values returned are listed in nanoseconds:
In the examples above, the precision is 500ms and the message delay is 3s.
Remember, `tendermint_consensus_quorum_prevote_delay` is listed in seconds.
```

File: docs/tools/debugging/proposer-based-timestamps-runbook.md [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+If the `tendermint_consensus_quorum_prevote_delay` value approaches the sum of `precision` and `message_delay`,
+then the value selected for these parameters is too small. Your application will
+need to be modified update the `SynchronyParams` to have larger values.
```
Comment (creachadair) [inline docs/tools/debugging/proposer-based-timestamps-runbook.md:193]:
```suggestion
need to be modified to update the `SynchronyParams` to have larger values.
```

File: docs/tools/debugging/proposer-based-timestamps-runbook.md [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+The `SynchronyParams` are `ConsensusParameters` which means they are set and updated
+by the application running alongside Tendermint. Updates to these parameters must
+be passed to the application during the `FinalizeBlock` ABCI method call.
+
+If the application was built using the CosmosSDK, then these parameters can be updated
+programatically using a governance proposal. For more information, see the [CosmosSDK
+documentation]().
+
+If the application does not implement a way to update the consensus parameters
+programatically, then it must be updated to do so. More information on updating
```
Comment (creachadair) [inline docs/tools/debugging/proposer-based-timestamps-runbook.md:206]:
```suggestion
programatically, then the application itself must be updated to do so. More information on updating
```

**Review by creachadair — COMMENTED** [review]
This looks really good. I made a bunch of minor copy-editing suggestions, but I think this is in pretty good shape.

File: docs/tools/debugging/proposer-based-timestamps-runbook.md [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+To use this runbook, you must be running a node that has the [prometheus metrics endpoint enabled](https://github.com/tendermint/tendermint/blob/841204c9a020cbc255fa117eca2ed6541b3786a7/docs/nodes/metrics.md#L5)
+and the Tendermint RPC endpoint enabled and accessible.
+
+It is strongly recommended to also run a Prometheus metrics collector to gather and
+analyze metrics from the Tendermint node.
+
+## Debugging a Single Node
+
+Use this runbook if you observe that your validator is frequently voting `nil` for a block that the rest
```
[the remark this answers is not in the document]
Comment (williambanfield) [reply docs/tools/debugging/proposer-based-timestamps-runbook.md]: Updated, please take a look. I kept some copy in this section as well to make it clear that this is somewhat different from the `Debugging a network` section.

[the remark this answers is not in the document]
Comment (creachadair) [reply docs/tools/debugging/proposer-based-timestamps-runbook.md]: Makes sense to me, thanks!

File: docs/tools/debugging/proposer-based-timestamps-runbook.md [hunk 7]
```diff
@@ -13,21 +13,25 @@ depending on the configuration of the validator's local clock.
...
+configuration of the Proposer-Based Timestamps [SynchronyParams](https://github.com/tendermint/tendermint/blob/master/spec/core/data_structures.md#synchronyparams).
+
+Use this runbook if you observe that validators are frequently voting `nil` for a block that the rest
+of the network votes for or if validators are frequently producing block proposals
+that are not voted for by the rest of the network.
 
 ## Requirements
 
-To use this runbook, you must be running a node that has the [prometheus metrics endpoint enabled](https://github.com/tendermint/tendermint/blob/841204c9a020cbc255fa117eca2ed6541b3786a7/docs/nodes/metrics.md#L5)
+To use this runbook, you must be running a node that has the [prometheus metrics endpoint enabled](https://github.com/tendermint/tendermint/blob/master/docs/nodes/metrics.md)
```
Comment (creachadair) [inline docs/tools/debugging/proposer-based-timestamps-runbook.md:24]:
```suggestion
To use this runbook, you must be running a node that has the [Prometheus metrics endpoint enabled](https://github.com/tendermint/tendermint/blob/master/docs/nodes/metrics.md)
```

**Review by creachadair — APPROVED** [review]

**Pull request merged**